syntax = "proto3";

package silenteight.learningstore.historicaldecision.v2.api;

import "google/api/field_behavior.proto";
import public "common.proto";


option java_package = "com.silenteight.proto.learningstore.historicaldecision.v2.api";
option java_outer_classname = "HistoricalDecisionLearningStoreExchangeProto";
option java_multiple_files = true;

// Message sent by Bridge to the Learning Engine queue. Related to Historical Decisions learning.
message HistoricalDecisionLearningStoreExchangeRequest {
  repeated Alert alerts = 1 [(google.api.field_behavior) = REQUIRED];
}

message Alert {
  string alert_id = 1 [(google.api.field_behavior) = REQUIRED]; // e.g. ALERT_ID_101
  string match_id = 2 [(google.api.field_behavior) = REQUIRED]; // e.g. MATCH_ID_011
  Watchlist watchlist = 3 [(google.api.field_behavior) = REQUIRED];
  AlertedParty alerted_party = 4 [(google.api.field_behavior) = REQUIRED];
  repeated Decision decisions = 5 [(google.api.field_behavior) = REQUIRED];
  repeated Discriminator discriminators = 6 [(google.api.field_behavior) = REQUIRED];
}

message AlertedParty {
  string id = 1 [(google.api.field_behavior) = REQUIRED]; // e.g. AP_ID_401
  string country = 2 [(google.api.field_behavior) = OPTIONAL]; // e.g. US
}
